In [ ]:
import numpy as np
import pandas as pd
import scipy.stats as stats

**Задание №1.** Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks): zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110], ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]. Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy Полученные значения должны быть равны. Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков, а затем с использованием функций из библиотек numpy и pandas.

In [ ]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

Воспользуемя формулой ковариации:
$$cov_{xy} = M(XY) - M(X) * M(Y), где: $$
где М - математическое ожидание

In [3]:
cov = np.mean(zp*ks) - np.mean(zp)*np.mean(ks)
print(cov)

9157.839999999997


In [4]:
np.cov(zp,ks, ddof=0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [5]:
r = cov/(np.std(zp)*np.std(ks))
print(r)

0.8874900920739158


In [7]:
np.corrcoef(zp, ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

**Задание №2.** Измерены значения IQ выборки студентов, обучающихся в местных технических вузах: 131, 125, 115, 122, 131, 115, 107, 99, 125, 111. Известно, что в генеральной совокупности IQ распределен нормально. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

Так как дисперсия неизвестна, используем распределение Стюдента.

$$\overline{\mathrm{x}} \pm \operatorname{t}_{\frac{\alpha}{2}}  \frac{\sigma}{\sqrt{n}}$$

По таблице для $\frac{\alpha}{2} = \frac{1 - 0.95}{2} = 0.025$ находим p = 0.975

In [18]:
alpha = 0.05
iq = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
n = len(iq)
t = stats.t.ppf(0.975, n - 1)

In [19]:
x_0 = np.mean(iq) - t*np.std(iq, ddof=1)/np.sqrt(n)
print(round(x_0, 2))

110.56


In [20]:
x_1 = np.mean(iq) + t*np.std(iq, ddof=1)/np.sqrt(n)
print(round(x_1, 2))

125.64


**Ответ:** Доверительный интервал для математического ожидания с надежностью 0.95 составляет: [110.56; 125.64]

**Задание №3.** Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

Для нахождения доверительного интервала для математического ожидания с надежностью 0.95 при известной σ генеральной совокупности используем следующую формулу:

$$X_{ср} +/- z_{\frac{\alpha}{2}} * \frac{\sigma}{\sqrt{n}}$$


In [22]:
alpha = 0.05
sigma = 5
n = 27
avg = 174.2

In [23]:
x_0 = avg - 1.96*sigma/np.sqrt(n)
print(round(x_0, 2))

172.31


In [24]:
x_1 = avg + 1.96*sigma/np.sqrt(n)
print(round(x_1, 2))

176.09


**Ответ:** Доверительный интервал для математического ожидания с надежностью 0.95 составляет: [172.31; 176.09]